src/main/java/com/odysseusinc/arachne/portal/service/impl/BaseArachneSecureServiceImpl.java

```diff
@@ -153,17 +156,12 @@ public BaseArachneSecureServiceImpl(UserStudyGroupedRepository userStudyGroupedR
         List<ParticipantRole> result = new LinkedList<>();
         if (submission != null) {
             Analysis analysis = submission.getSubmissionGroup().getAnalysis();
-            if (analysis != null && analysis.getStudy() != null) {
-                result = getRolesByStudy(user, analysis.getStudy());
-                final DataNode dataNode = submission.getDataSource().getDataNode();
-                if (!DataNodeUtils.isDataNodeOwner(dataNode, user.getId())) {
-                    // check if we are not owner - delete owner role O_O
-                    result.removeIf(ParticipantRole.DATA_SET_OWNER::equals);
-                }
-            } else if (analysis != null) {
```

Comment:
Could you check, when this code was added and what was fixed by it? And verify that we do not bring back the issue